File: api/send_report.js
const { Resend } = require('resend');

const resend = new Resend(process.env.RESEND_API_KEY);
const dest_mail = process.env.DEST_MAIL;

export default async function handler(req, res) {
    // 1. Configuration des Headers CORS
    res.setHeader('Access-Control-Allow-Origin', 'https://ilesduponant.github.io');
    res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    // 2. Gestion de la requête de pré-vérification (Preflight)
    if (req.method === 'OPTIONS') {
        return res.status(200).end();
    }

    if (req.method !== 'POST') {
        return res.status(405).json({ error: 'Méthode non autorisée' });
    }

    try {
        const { nomCli, prenomCli, npCli, ile, no_dossier, zip_data } = req.body;

        // 3. Envoi via Resend
        const data = await resend.emails.send({
            from: 'Branchement <onboarding@resend.dev>',
            to: [dest_mail],
            subject: `BRANCHEMENT ${nomCli} - ${no_dossier}`,
            html: `<p>Informations pour l'étude de branchement <strong>${npCli}</strong>.</p><p>Dossier n°${no_dossier}</p>`,
            attachments: [
                {
                    filename: `Raccordement_${nomCli}_${ile}.zip`,
                    content: zip_data, //base64
                },
            ],
        });

        return res.status(200).json({ status: 'success', id: data.id });
    } catch (error) {
        console.error("Erreur Resend:", error);
        return res.status(500).json({ error: error.message });
    }
}

File: api/send_report.py
from http.server import BaseHTTPRequestHandler
import json
import os
import resend

class handler(BaseHTTPRequestHandler):
    def do_OPTIONS(self):
        # Indispensable pour autoriser les requêtes provenant de ton site GitHub
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'POST, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()

    def do_POST(self):
        content_length = int(self.headers['Content-Length'])
        post_data = self.rfile.read(content_length)
        
        try:
            body = json.loads(post_data)
            resend.api_key = os.environ.get("RESEND_API_KEY")
            
            nom_client = body.get('nom_client', 'Inconnu')
            no_dossier = body.get('no_dossier', 'Sans Numero')
            zip_data = body.get('zip_data')

            # Envoi via Resend
            params = {
                "from": "Rapports <onboarding@resend.dev>",
                "to": ["eryuv1829@gmail.com"],
                "subject": f"Rapport: {no_dossier} - {nom_client}",
                "html": f"<strong>Nouveau rapport reçu</strong><br>Dossier: {no_dossier}<br>Client: {nom_client}",
                "attachments": [
                    {
                        "filename": f"photos_{no_dossier}.zip",
                        "content": list(zip_data) if isinstance(zip_data, bytes) else zip_data
                    }
                ]
            }
            
            resend.Emails.send(params)

            self.send_response(200)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({"status": "success"}).encode())

        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(json.dumps({"error": str(e)}).encode())